### PR TITLE
BUG: Fixes plotting with nullable integers (#32073)

### LINKED ISSUE #32073: Plotting Int64 columns with nulled integers (NAType) fails
#### Code Sample, a copy-pastable example if possible

```
import pandas as pd
import numpy as np

df = pd.DataFrame({'A': [1, 2, 3, 4, 5], 'B': [7, 5, np.nan, 3, 2]})
df.plot(x='A', y='B')
df = df.astype('Int64')
df.plot(x='A', y='B')
```
#### Problem description

The first plotting command works, the second throws the error message

> TypeError: float() argument must be a string or a number, not 'NAType'

#### Expected Output

NAType should be treated the same way as numpy nan in plotting. Maybe transformed on the fly?

(I'm unsure if this is a pandas, a numpy, or a matplotlib issue, I'm starting here)

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.6.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
machine          : AMD64
processor        : Intel64 Family 6 Model 79 Stepping 1, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : en
LOCALE           : None.None

pandas           : 1.0.1
numpy            : 1.18.1
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 20.0.2
setuptools       : 45.2.0.post20200209
Cython           : 0.29.15
pytest           : None
hypothesis       : None
sphinx           : 2.4.1
blosc            : None
feather          : None
xlsxwriter       : 1.2.7
lxml.etree       : 4.5.0
html5lib         : 1.0.1
pymysql          : None
psycopg2         : None
jinja2           : 2.11.1
IPython          : 7.12.0
pandas_datareader: None
bs4              : 4.8.2
bottleneck       : None
fastparquet      : 0.3.3
gcsfs            : None
lxml.etree       : 4.5.0
matplotlib       : 3.1.3
numexpr          : None
odfpy            : None
openpyxl         : 3.0.3
pandas_gbq       : None
pyarrow          : 0.16.0
pytables         : None
pytest           : None
pyxlsb           : None
s3fs             : None
scipy            : 1.4.1
sqlalchemy       : 1.3.13
tables           : None
tabulate         : None
xarray           : None
xlrd             : 1.2.0
xlwt             : None
xlsxwriter       : 1.2.7
numba            : 0.48.0

</details>

### PR DESCRIPTION
**Pull request by rkc007**
- [x] closes #32073
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by rkc007** [comment]
This is the first time I am contributing to pandas open source project. Please let me know if I can improve anything on this commit.

File: doc/source/whatsnew/v1.2.0.rst [hunk 1]
```diff
@@ -681,6 +681,7 @@ Plotting
   indexed by a :class:`TimedeltaIndex` with a fixed frequency when x-axis lower limit was greater than upper limit (:issue:`37454`)
 - Bug in :meth:`DataFrameGroupBy.boxplot` when ``subplots=False``, a KeyError would raise (:issue:`16748`)
 - Bug in :meth:`DataFrame.plot` and :meth:`Series.plot` was overwriting matplotlib's shared y axes behaviour when no sharey parameter was passed (:issue:`37942`)
+- Bug in :meth:`DataFrame.plot` to handle nullable integers (:issue:`32073`)
```
Comment (arw2019) [inline doc/source/whatsnew/v1.2.0.rst:684]: We want to be more specific:
```
Bug in :meth:`DataFrame.plot` was raising a ``TypeError`` with ``ExtensionDtype`` columns (:issue:`32073`)
```

File: pandas/plotting/_matplotlib/core.py [hunk 2]
```diff
@@ -423,13 +425,19 @@ def _compute_plot_data(self):
         if is_empty:
             raise TypeError("no numeric data to plot")
 
-        # GH25587: cast ExtensionArray of pandas (IntegerArray, etc.) to
-        # np.ndarray before plot.
-        numeric_data = numeric_data.copy()
-        for col in numeric_data:
-            numeric_data[col] = np.asarray(numeric_data[col])
+        def convert_to_ndarray(data):
```
Comment (arw2019) [inline pandas/plotting/_matplotlib/core.py:428]: you could use `pd.to_numeric` here I think. Illustrative example:
``` python
In [8]: s = pd.Series([1, 2, 3, pd.NA], dtype="Int64")

In [9]: pd.to_numeric(s.to_numpy(), downcast="integer")
Out[9]: array([ 1.,  2.,  3., nan])
```

File: pandas/tests/plotting/frame/test_frame.py [hunk 3]
```diff
@@ -148,6 +148,23 @@ def test_plot(self):
...
+            {
+                "A": [1, 2, 3, 4, 5],
+                "B": [7, 5, np.nan, 3, 2],
+                "C": pd.to_datetime(dates, format="%Y"),
+                "D": pd.to_datetime(dates, format="%Y", utc=True),
+            }
+        )
+
+        _check_plot_works(df.plot, x="A", y="B")
+        _check_plot_works(df[["A", "B"]].astype("Int64").plot, x="A", y="B")
```
Comment (arw2019) [inline pandas/tests/plotting/frame/test_frame.py:164]: I think we'd rather to specify the dtype in the dataframe construction

**Review by arw2019 — CHANGES_REQUESTED** [review]
Thanks @rkc007 for the PR!

Some comments. Also take a look at the CI (we standardize the use of `pd.DataFrame`/`DataFrame` within a file so having both errors)

Let us know if you have any questions!

File: doc/source/whatsnew/v1.2.0.rst (hunk 1 above)
Comment (rkc007) [reply doc/source/whatsnew/v1.2.0.rst:684]: Thank you @arw2019 for your comments. I will add these changes to the next commit.

File: pandas/plotting/_matplotlib/core.py (hunk 2 above)
Comment (rkc007) [reply pandas/plotting/_matplotlib/core.py:428]: Thank you @arw2019 for the example. I think I didn't get your point here. Can you please explain it a bit more?

File: pandas/tests/plotting/frame/test_frame.py (hunk 3 above)
Comment (rkc007) [reply pandas/tests/plotting/frame/test_frame.py:164]: Sure. Thank you for the suggestion. I will rectify this in the next commit.

File: pandas/plotting/_matplotlib/core.py (hunk 2 above)
Comment (ivanovmg) [inline pandas/plotting/_matplotlib/core.py:428]: Consider moving ``convert_to_ndarray`` out of the scope of the function ``_compute_plot_data``.

File: pandas/tests/plotting/frame/test_frame.py [hunk 4]
```diff
@@ -148,6 +148,23 @@ def test_plot(self):
         result = ax.axes
         assert result is axes[0]
 
+    def test_nullable_int_plot(self):
+        # GH 32073
+        dates = ["2008", "2009", None, "2011", "2012"]
+        df = pd.DataFrame(
+            {
+                "A": [1, 2, 3, 4, 5],
+                "B": [7, 5, np.nan, 3, 2],
```
Comment (ivanovmg) [inline pandas/tests/plotting/frame/test_frame.py:157]: What about having a nullable value in column "A" (x-axis)?

**Review by ivanovmg — CHANGES_REQUESTED** [review]
@rkc007, please see my comments below.

File: pandas/plotting/_matplotlib/core.py (hunk 2 above)
Comment (jreback) [reply pandas/plotting/_matplotlib/core.py:428]: no don't use `pd.to_numeric` we already know the type and its an internal function

Comment (jreback) [reply pandas/plotting/_matplotlib/core.py:428]: yes pls do this

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/plotting/_matplotlib/core.py [hunk 5]
```diff
@@ -423,13 +425,19 @@ def _compute_plot_data(self):
...
+            # GH32073: cast to float if values contain nulled integers
+            if is_integer_dtype(data.dtype) and is_extension_array_dtype(data.dtype):
```
Comment (jreback) [inline pandas/plotting/_matplotlib/core.py:430]: actually we want to also catch / test nullable floats as well

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v1.2.0.rst (hunk 1 above)
Comment (rkc007) [reply doc/source/whatsnew/v1.2.0.rst]: @arw2019 I added the changes to the latest commit. Can you please review it again ?

File: pandas/tests/plotting/frame/test_frame.py (hunk 3 above)
Comment (rkc007) [reply pandas/tests/plotting/frame/test_frame.py]: @arw2019 I added the changes to the latest commit. Can you please review it again ?

File: pandas/plotting/_matplotlib/core.py (hunk 2 above)
Comment (rkc007) [reply pandas/plotting/_matplotlib/core.py]: @jreback I added the changes to the latest commit. Can you please review it again ?

File: pandas/tests/plotting/frame/test_frame.py (hunk 4 above)
Comment (rkc007) [reply pandas/tests/plotting/frame/test_frame.py]: @ivanovmg I added the changes to the latest commit. Can you please review it again ?

File: pandas/plotting/_matplotlib/core.py (hunk 5 above)
Comment (rkc007) [reply pandas/plotting/_matplotlib/core.py]: @jreback I fixed this in the latest commit.

File: pandas/plotting/_matplotlib/core.py [hunk 6]
```diff
@@ -383,6 +386,21 @@ def result(self):
             else:
                 return self.axes[0]
 
+    def _convert_to_ndarray(self, data):
+        # data = self.data
```
Comment (ivanovmg) [inline pandas/plotting/_matplotlib/core.py:390]: Please delete this comment.

**Review by ivanovmg — APPROVED** [review]
One nitpick about the unnecessary comment.
Otherwise looks good to me.

File: pandas/plotting/_matplotlib/core.py (hunk 6 above)
Comment (rkc007) [reply pandas/plotting/_matplotlib/core.py]: Thanks for pointing this out @ivanovmg. 
Deleted the comment in the latest commit.

**Review by jreback — APPROVED** [review]
looks good. can you merge master and ping on green.

**Comment by rkc007** [comment]
@jreback the branch has the latest code and all the checks passed.

**Comment by jreback** [comment]
thanks @rkc007

**Pull request merged**